src/Microsoft.OData.Client/DataServiceClientFormat.cs

```diff
@@ -175,12 +175,14 @@ internal void SetRequestAcceptHeaderForCount(HeaderCollection headers)
         }
 
         /// <summary>
-        /// Sets the value of the Accept header for a count request (will set it to 'multipart/mixed').
+        /// Sets the value of the Accept header for a batch request
+        /// Will set it to 'multipart/mixed' for a multipart batch request
+        /// Will set it to 'application/json;odata.metadata=minimal' for a json batch request

```

Comment:
It's important to note that multipart batch request header does not automatically imply multipart batch response (same for json batch request & json batch response). It is possible to send application/json content-type and receive multipart response. In fact for a default webapi setup sending a json batch response returns a multipart response unless you explicitly set accept headers as json as well.
Maybe we could have it as the default behaviour but then give the user the option to [un]set it
cc @mikepizzo for confirmation on whether this is by design or an oversight on the odata service side

Reply:
It is certainly possible for a mulitpart batch request to return a json batch response and vice-versa.  I *thought* we specified that the default response was the same as the request, but I didn't find that doing a quick search and it sounds from the comment above like that's not the behavior of the webapi stack.

For the client, I would suggest that we always specify both Content-Type and Accept headers; in the UseJson case we would set *both* to application/json, and in the multipart scenario set them both to multipart. I don't think there's a benefit in providing an option for the data service client to send a request as multipart and get back JSON response, or vice-versa.

Reply:
#Resolved